pinot-common/src/main/java/com/linkedin/pinot/common/request/transform/TransformExpressionTree.java

```diff
@@ -31,25 +30,47 @@
  * <p> - Transform function in non-leaf nodes is applied to its children nodes. </p>
  */
 public class TransformExpressionTree {
-  private final String _column;
-  private final String _literal;
+
+  // Enum for expression represented by the tree.
+  public enum ExpressionType {
+    FUNCTION,
+    IDENTIFIER,
+    LITERAL
+  }
+
+  private final String _expression;
   private final String _transformName;
   private final List<TransformExpressionTree> _children;
+  private final ExpressionType _expressionType;
 
   /**
    * Constructor for the class
-   * @param column Column on which to apply the transform.
-   * @param literal A constant value, to be used as an argument for the transform function
-   * @param transformName Name of transform function to apply
+   * @param root Root AstNode for the tree
    * @param children Children nodes for the current node
    */
-  public TransformExpressionTree(String column, String literal, String transformName,
-      List<TransformExpressionTree> children) {
-    _column = column;
-    _literal = literal;
-    _transformName = transformName;
+  public TransformExpressionTree(AstNode root, List<TransformExpressionTree> children) {
```

Comment:
The buildTree method is a public static method that the compiler uses to build the expression tree, it cannot be removed.

Each tree (node) needs to know its children somehow. The current code is passing an array  list (reference), that gets filled later. On second thought, a better approach would be to collect all children and set them via a setter.